src/index/index.c

```diff
@@ -0,0 +1,265 @@
+#include "index.h"
+#include "index_type.h"
+
+int compareNodes(const void *a, const void *b) {
+  return (GrB_Index)a - (GrB_Index)b;
+}
+
+int compareStrings(void *a, void *b, void *ctx) {
+  return strcmp(((SIValue*)a)->stringval, ((SIValue*)b)->stringval);
+}
+
+// TODO this seems inefficient
+int compareNumerics(void *p1, void *p2, void *ctx) {
+  double a, b;
+  SIValue_ToDouble(p1, &a);
+  SIValue_ToDouble(p2, &b);
+  return a - b;
+}
+
+/*
+ * The index must maintain its own copy of the indexed SIValue
+ * so that it becomes outdated but not broken by updates to the property.
+ */
+void cloneKey(void **property) {
```

Comment:
I agree. I think that I will create typedefs for  `skiplistVal` and `skiplistKey` types. Currently, Index keys are always SIValues, and Index values are always Node IDs. 

This should simplify the logic in a lot of areas.

In the future, we may want to extend functionality to index edges as well as nodes and/or to support composite keys, but I think it would be better to make the necessary adjustments at that time rather than leaving the door open with all of this `void *` casting right now.